code/DynamicCache.php

```diff
@@ -384,6 +393,10 @@ public function run($url)
         // This prevents a new user's security token from being regenerated incorrectly
         $_SESSION['SecurityID'] = SecurityToken::getSecurityID();
 
+        // Set the stage of the website
+        // This is normally called in VersionedRequestFilter.
+        Versioned::choose_site_stage();
```

Comment:
Also can_choose_site_stage isn't called, so this code could introduce a security vulnerability if called in isolation.